client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx

```diff
@@ -138,7 +140,7 @@ export const PreMigrationScreen: React.FunctionComponent< PreMigrationProps > =
 	// We want to record the tracks event, so we use the same condition as the one in the render function
 	// This should be better handled by using a state after the refactor
 	useEffect( () => {
-		if ( ! requiresPluginUpdate && isTargetSitePlanCompatible ) {
+		if ( siteCanMigrate === true && isTargetSitePlanCompatible ) {
```

Comment:
After merging https://github.com/Automattic/wp-calypso/pull/83088, we can remove this entire `useEffect` block and move triggering the "calypso_site_migration_ready_screen" event to the MigrationReady component.

Reply:
That's what I'm thinking as well, I'll hold this PR for now until #83088 is merged!

Reply:
@bogiii I've move the useEffect into it's own component, could you review it again?